src/lib/shadow-shim-helper-rs/src/syscall_types.rs

```diff
@@ -0,0 +1,307 @@
+/// Represents a pointer to a virtual address in plugin memory.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[repr(C)]
+pub struct PluginPtr {
+    val: usize,
+}
```

Comment:
Maybe add a comment here that says `SysCallArgs` requires this to be `repr(C)` since `transmute::<u64, PluginPtr>()` must be valid, so that we don't accidentally remove the `repr(C)` once we no longer access it from C.

Reply:
Added static assertions and comments near the `SysCallArgs` definition - I think it makes more sense there.